pkg/chartverifier/checks/registry.go

```diff
@@ -82,31 +108,29 @@ type CheckOptions struct {
 type CheckFunc func(options *CheckOptions) (Result, error)
 
 type Registry interface {
-	Get(name string) (Check, bool)
-	Add(check Check) Registry
-	AllChecks() []string
+	Get(id CheckId) (Check, bool)
+	Add(name CheckName, version Version, checkFunc CheckFunc) Registry
+	AllChecks() DefaultRegistry
 }
 
-type defaultRegistry map[string]Check
+type DefaultRegistry map[CheckId]Check
```

Comment:
do we really need here map and not slice? Also, would not clearer if we call this type `Checks`?

Reply:
I think we do, a check is identified now by the check name and the version which is encapsulated in CheckId. The name could be a single string  combination of name and version but that adds complication when processing the -x and -e  flags which only specify a check name. Also if you do a slice if we want to get a check based on its name and version we would have to loop though the slice instead of getting the value based on a key.